pkg/service/json.go

```diff
@@ -144,13 +145,12 @@ func ParseJSONBody(body io.Reader, obj interface{}) error {
 
 	err := json.NewDecoder(body).Decode(&obj)
 	if err != nil {
-		switch err := err.(type) {
-		case *json.UnmarshalTypeError:
-			return NewInvalidParameterError(err.Field, fmt.Sprintf("must be %s", primitiveTypeToDisplayName(err.Type)))
-		default:
-			if err != io.EOF {
-				return NewInvalidRequestError("Invalid request body")
-			}
+		var unmarshalTypeErr *json.UnmarshalTypeError
+		if errors.As(err, &unmarshalTypeErr) {
+			return NewInvalidParameterError(unmarshalTypeErr.Field, fmt.Sprintf("must be %s", primitiveTypeToDisplayName(unmarshalTypeErr.Type)))
+		}
+		if !errors.Is(err, io.EOF) {
+			return NewInvalidRequestError("Invalid request body")
```

Comment:
Same comment as above: Can you log err above this return for easier debugging? Something like `log.Error().Err(err).Msgf("json: invalid request body: ParseJSONBody")`